ext/reflection/php_reflection.c

```diff
@@ -682,6 +682,26 @@ static int format_default_value(smart_str *str, zval *value) {
 			format_default_value(str, zv);
 		} ZEND_HASH_FOREACH_END();
 		smart_str_appendc(str, ']');
+	} else if (Z_TYPE_P(value) == IS_OBJECT) {
+		// Look through the global constants, GH-15902
+		zend_constant *constant;
+		zend_string *name = NULL;
+
+		ZEND_HASH_MAP_FOREACH_PTR(EG(zend_constants), constant) {
```

Comment:
I think instead of trying to be accurate about where the object is coming from, it should be enough to simply dump `object(Foo)` or whatnot (not sure what the best format would be). We cannot accurately detect it anyway, e.g. when there's indirection through some other constant.

```php
define('FOO', new stdClass);

class C {
	public const FOO = FOO;
	public const BAR = self::FOO;
}
```

Reply:
I'm a bit confused by the `object(Foo)` - is that meant to represent the value, e.g. the stdClass? We may not be able to 100% detect it, but at least this is closer than `callable`, `__CLASS__`, or `...`
If we don't want to show the original AST, then we should also change the behavior of the unevaluated AST to evaluate it, i.e. we shouldn't show the AST in some cases and the value in others.

Reply:
> i.e. we shouldn't show the AST in some cases and the value in others.

The "problem" is that the AST goes away when it is evaluated. This doesn't need to be super accurate, IMO. E.g. for arrays evaluated from constants, this would already not be accurate.

Reply:
I realize the AST goes away when evaluated, and I am proposing that we intentionally take advantage of that, so that unevaluated AST is shown the same as the evaluated result (where the output is the evaluated result), since in this patch I wanted to have them shown the same where the output is the original AST, and my understanding is that you are objecting to that

Reply:
Right, what I'm objecting to is the complexity/cost of iterating all constants, when it isn't reliable anyway. For example, see how for arrays, we already dump the array value instead of the constant name after it has been evaluated. https://3v4l.org/F364P

Reply:
Okay, I changed it to just show the type of the object